improver/cli/__init__.py

```diff
@@ -444,6 +464,31 @@ def unbracket(args):
     return outargs
 
 
+class TimeIt:
+    def __init__(self, verbose=False):
+        self._verbose = verbose
+        self._elapsed = None
+        self._start = None
+
+    def __enter__(self):
+        self._start = time.perf_counter()
+        return self
+
+    def __exit__(self, *args):
+        self._elapsed = time.perf_counter() - self._start
+        if self._verbose:
+            print(str(self))
+
+    @property
+    def elapsed(self):
+        """Return elapsed time in seconds."""
+        return self._elapsed
```

Comment:
No circumstance you'd want to use query during execution?

Reply:
Sorry @LaurenceBeard, can you clarify what you mean?
I thought it useful to provide access to the elapsed time as a numeric value (perhaps when it comes to logging).
Are you suggesting removing this property? (we don't use it right now)

Reply:
Nope, just pointing out that it'll be None until exit, was checking there was no case for this (it appears not!)